Samples/ObservableScreen/Sources/CounterWorkflow.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+import Workflow
+import WorkflowSwiftUI
+import WorkflowUI
+
+struct CounterWorkflow: Workflow {
+    // Dependencies from parent.
+    let info: CounterInfo
+    let resetToken: ResetToken
+    let initialValue: Int
+    let maxValue: Int?
+
+    @ObservableState
+    struct State {
+        private var _count: Int
+        
+        var count: Int {
+            get { _count }
+            set {
+                if let maxValue, newValue > maxValue {
```

Comment:
Is there a concern that the logic for protecting state can be located in a setter as well as in an action? I mean, this could probably have been done prior to SwiftUI, but this state is mutable from the SwiftUI view because it's `@ObservableState`, so will be much easier to do.

Reply:
Well, it's not exactly the setter that's the concern, but because accessible properties get a "free" mutation action, there is definitely a gotcha where you might declare an action and forget to use it. It'll be easily discoverable if your action does anything, but might still be an annoying lesson to learn.